src/components/Tasks/TaskListItem.tsx

```diff
@@ -36,26 +36,40 @@ interface PropsType {
   entry: TaskType;
   onClick: (task: TaskType) => void;
   onSave: (item: PimType, journalUid: string, originalItem?: PimType) => Promise<void>;
+  onUpdate: (message: string) => void;
 }
 
 export default React.memo(function TaskListItem(props: PropsType) {
   const {
     entry: task,
     onClick,
     onSave: save,
+    onUpdate: update,
   } = props;
   const title = task.title;
 
-  function toggleComplete(_e: React.ChangeEvent<HTMLInputElement>, checked: boolean) {
+  async function toggleComplete(_e: React.ChangeEvent<HTMLInputElement>, checked: boolean) {
     const clonedTask = task.clone();
     clonedTask.status = checked ? TaskStatusType.Completed : TaskStatusType.NeedsAction;
-    save(clonedTask, (task as any).journalUid, task);
+
+    const nextTask = checked ? task.getNextOccurence() : null;
+
+    await save(clonedTask, (task as any).journalUid, task);
+
+    if (nextTask) {
+      await save(nextTask, (task as any).journalUid);
+      update(`${nextTask.title} rescheduled for ${formatDate(nextTask.startDate ?? nextTask.dueDate)}`);
+    }
   }
 
+  const dueDateText = task.dueDate ? `Due ${formatDate(task.dueDate)}` : '';
+  const freqText = task.rrule ? `(repeats ${task.rrule.freq.toLowerCase()})` : '';
+  const secondaryText = `${dueDateText} ${freqText}`;
```

Comment:
This looks odd if there's no due date...

Reply:
Really? I actually like how it's displayed, I think it gives context for the task. Should I remove the parenthesis if there is no due date? Or just remove it entirely if there is no due date?

Reply:
Sorry, let me clarify, I just realise that my message was really bad.
When a task has no dueDate set, there's a redundant space before the `freqText` which looks odd.

Reply:
Ah ok I can make sure that's removed if not necessary

Reply:
Looking at this in my browser (Firefox) and the leading white-space is removed when the component is rendered. Where are you seeing this issue?

Reply:
I can't see it either. Never mind then. :|
Sorry for the noise.